src/metabase/query_processor/util.clj

```diff
@@ -84,23 +85,41 @@
        x))
    x))
 
+(defn- remove-lib-uuids
+  "Two queries should be the same even if they have different :lib/uuids, because they might have both been converted
+  from the same legacy query."
+  [x]
+  (walk/postwalk
+   (fn [x]
+     (if (map? x)
+       (dissoc x :lib/uuid)
+       x))
+   x))
```

Comment:
Query hash is now done on pMBQL queries, I had to remove `:lib/uuid` to not break everything because one queries converted from legacy two times will get two different hashes otherwise.